react/features/base/participants/functions.js

```diff
@@ -45,3 +50,65 @@ function _getParticipants(participantsOrGetState) {
 
     return participants || [];
 }
+
+/**
+ * Returns the URL of the image for the avatar of a particular participant
+ * identified by their id and/or e-mail address.
+ *
+ * @param {string} participantId - Participant's id.
+ * @param {Object} [options] - The optional arguments.
+ * @param {string} [options.avatarUrl] - Participant's avatar url.
+ * @param {string} [options.email] - Participant's email.
+ * @param {string} [options.avatarId] - Participant's avatar id.
+ * @returns {string} The URL of the image for the avatar of the participant
+ * identified by the specified participantId and/or email.
+ *
+ * @public
+ */
+export function getAvatarURL(participantId, options = {}) {
+    if (typeof config === 'object' && config.disableThirdPartyRequests) {
+        return 'images/avatar2.png';
+    }
+    const { avatarId, avatarUrl, email } = options;
+
+    // The priority is url, email and lowest is avatarId
+    if (avatarUrl) {
+        return avatarUrl;
+    }
+
+    let avatar = null;
+
+    if (email) {
+        avatar = email;
+    } else {
+        avatar = avatarId;
+    }
+
+    // If the ID looks like an email, we'll use gravatar.
+    // Otherwise, it's a random avatar, and we'll use the configured
+    // URL.
+    const random = !avatar || avatar.indexOf('@') < 0;
+
+    if (!avatar) {
+        avatar = participantId;
+    }
+
+    avatar = MD5.hexdigest(avatar.trim().toLowerCase());
+
+    let urlPref = null;
+    let urlSuf = null;
+
+    if (!random) {
+        urlPref = 'https://www.gravatar.com/avatar/';
+        urlSuf = '?d=wavatar&size=200';
+    } else if (random && typeof interfaceConfig === 'object'
+        && interfaceConfig.RANDOM_AVATAR_URL_PREFIX) {
+        urlPref = interfaceConfig.RANDOM_AVATAR_URL_PREFIX;
+        urlSuf = interfaceConfig.RANDOM_AVATAR_URL_SUFFIX;
+    } else {
+        urlPref = 'https://api.adorable.io/avatars/200/';
+        urlSuf = '.png';
+    }
```

Comment:
The naming here is very misleading. !random = gravatar, random = custom prefixes and default = adorable.io..Let's rename the "random" variable to "custom" maybe?